### PR TITLE
Fix wrong handling of nuraft config logs in HA durability

### PR DESCRIPTION
**Pull request by antoniofilipovic**
This diff fixes the issue with the wrong handling of the Nuraft config logs in durability, where they were parsed as text although they are serialized in Nuraft differently.

### CODE REVIEW

**Comment by antoniofilipovic** [comment]
Git commit description, explain the changes you made here

---
__*Leave above in PR description, copy the below into a comment*__
___


### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

File: src/coordination/coordinator_state_machine.cpp [hunk 1]
```diff
@@ -181,47 +181,60 @@ auto CoordinatorStateMachine::SerializeUpdateUUIDForInstance(InstanceUUIDUpdate
...
 
 auto CoordinatorStateMachine::commit(ulong const log_idx, buffer &data) -> ptr<buffer> {
+  auto const return_successful_commit = [this, &log_idx]() {
+    last_committed_idx_ = log_idx;
+    logger_.Log(nuraft_log_level::TRACE, fmt::format("Last commit index: {}", last_committed_idx_));
+    ptr<buffer> ret = buffer::alloc(sizeof(log_idx));
+    buffer_serializer bs_ret(ret);
+    bs_ret.put_u64(log_idx);
+    return ret;
+  };
```
Comment (as51340) [inline src/coordination/coordinator_state_machine.cpp:226]: why this change, in both cases (if-else) we do the same?

File: src/coordination/coordinator_state_machine.cpp [hunk 2]
```diff
@@ -181,47 +181,60 @@ auto CoordinatorStateMachine::SerializeUpdateUUIDForInstance(InstanceUUIDUpdate
...
 
-auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::pair<TRaftLog, RaftLogAction> {
+auto CoordinatorStateMachine::DecodeLog(buffer &data) -> std::optional<std::pair<TRaftLog, RaftLogAction>> {
   buffer_serializer bs(data);
-  auto const json = nlohmann::json::parse(bs.get_str());
+  std::string const log_str = bs.get_str();
+  if (log_str.empty()) {
+    return std::nullopt;  // this is returned only when the log is configuration log
+  }
+  auto const json = nlohmann::json::parse(log_str);
```
Comment (as51340) [inline src/coordination/coordinator_state_machine.cpp:190]: Does this mean we cannot create log with empty string or? Do we always have some string

**Review by as51340 — CHANGES_REQUESTED** [review]

File: src/coordination/coordinator_state_machine.cpp (hunk 2 above)
Comment (antoniofilipovic) [reply src/coordination/coordinator_state_machine.cpp:190]: Actually I don't need this but it was used as a safety check if data is empty we have nothing to decode. But that case is handled in commit function in this file so I will remove this

File: src/coordination/coordinator_state_machine.cpp (hunk 1 above)
Comment (antoniofilipovic) [reply src/coordination/coordinator_state_machine.cpp:226]: You are right, this can be same part without func, I thought just of isolating same part in function

File: src/coordination/coordinator_state_machine.cpp (hunk 2 above)
Comment (antoniofilipovic) [reply src/coordination/coordinator_state_machine.cpp:190]: I changed this part as I figured out we have commit config function

**Review by as51340 — APPROVED** [review]

**Comment by antoniofilipovic** [comment]
@kgolubic RN: 
This diff fixes the issue with the wrong handling of the Nuraft config logs in durability, where they were parsed from RocksDB as text although they are serialized in Nuraft binary format.
